ICE: `span context mismatch`

<!--
[31mICE[0m: Rustc ./meta-variable-misuse.rs '-Cinstrument-coverage -ooutputfile -Zdump-mir-dir=dir' 'thread 'rustc' ($TID) panicked at compiler/rustc_mir_transform/src/coverage/spans.rs:56:13: 'span context mismatch: body_span=/tmp/im/meta-variable-misuse.rs:7:54: 7:60 (#5), covspan.span=/tmp/im/meta-variable-misuse.rs:7:56: 7:58 (#6)'', 'thread 'rustc' ($TID) panicked at compiler/rustc_mir_transform/src/coverage/spans.rs:56:13: 'span context mismatch: body_span=/tmp/im/meta-variable-misuse.rs:7:54: 7:60 (#5), covspan.span=/tmp/im/meta-variable-misuse.rs:7:56: 7:58 (#6)''
File: /tmp/im/meta-variable-misuse.rs
-->

auto-reduced (treereduce-rust):
````rust
//@compile-flags: -Cinstrument-coverage
macro_rules! foo {
    ($($m:ident $($f:ident $v:tt)+),*) => {
        $($(macro_rules! $f { () => { $v } })+)*
        $(macro_rules! $m { () => { $(fn $f() -> i32 { $v })+ } })*
    }
}

foo!(m a 1 b 2, n c 3);
m!();
n!();

````

original:
````rust
//@ run-pass
#![deny(meta_variable_misuse)]

macro_rules! foo {
    ($($m:ident $($f:ident $v:tt)+),*) => {
        $($(macro_rules! $f { () => { $v } })+)*
        $(macro_rules! $m { () => { $(fn $f() -> i32 { $v })+ } })*
    }
}

foo!(m a 1 b 2, n c 3);
m!();
n!();

macro_rules! no_shadow {
    ($x:tt) => { macro_rules! bar { ($x:tt) => { 42 }; } };
}
no_shadow!(z);

macro_rules! make_plus {
    ($n: ident $x:expr) => { macro_rules! $n { ($y:expr) => { $x + $y }; } };
}
make_plus!(add3 3);

fn main() {
    assert_eq!(a!(), 1);
    assert_eq!(b!(), 2);
    assert_eq!(c!(), 3);
    assert_eq!(a(), 1);
    assert_eq!(b(), 2);
    assert_eq!(c(), 3);
    assert_eq!(bar!(z:tt), 42);
    assert_eq!(add3!(9), 12);
}

````

Version information
````
rustc 1.92.0-nightly (a0f398e89 2025-10-04)
binary: rustc
commit-hash: a0f398e89df9767c93c81cd58d44fdba071258a8
commit-date: 2025-10-04
host: x86_64-unknown-linux-gnu
release: 1.92.0-nightly
LLVM version: 21.1.2

````

Possibly related line of code:
https://github.com/rust-lang/rust/blob/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_mir_transform/src/coverage/spans.rs#L50-L62

Command:
`/home/matthias/.rustup/toolchains/alt-master/bin/rustc -Cinstrument-coverage`

<details><summary><strong>Program output</strong></summary>
<p>

```
warning: unused macro definition: `a`
 --> /tmp/icemaker_global_tempdir.X6mzSRUkIobi/rustc_testrunner_tmpdir_reporting.K2C7radLLRcJ/mvce.rs:8:8
  |
8 | foo!(m a 1 b 2, n c 3);
  |        ^
  |
  = note: `#[warn(unused_macros)]` (part of `#[warn(unused)]`) on by default

warning: unused macro definition: `b`
 --> /tmp/icemaker_global_tempdir.X6mzSRUkIobi/rustc_testrunner_tmpdir_reporting.K2C7radLLRcJ/mvce.rs:8:12
  |
8 | foo!(m a 1 b 2, n c 3);
  |            ^

warning: unused macro definition: `c`
 --> /tmp/icemaker_global_tempdir.X6mzSRUkIobi/rustc_testrunner_tmpdir_reporting.K2C7radLLRcJ/mvce.rs:8:19
  |
8 | foo!(m a 1 b 2, n c 3);
  |                   ^

error[E0601]: `main` function not found in crate `mvce`
  --> /tmp/icemaker_global_tempdir.X6mzSRUkIobi/rustc_testrunner_tmpdir_reporting.K2C7radLLRcJ/mvce.rs:10:6
   |
10 | n!();
   |      ^ consider adding a `main` function to `/tmp/icemaker_global_tempdir.X6mzSRUkIobi/rustc_testrunner_tmpdir_reporting.K2C7radLLRcJ/mvce.rs`


thread 'rustc' (3331399) panicked at compiler/rustc_mir_transform/src/coverage/spans.rs:56:13:
span context mismatch: body_span=/tmp/icemaker_global_tempdir.X6mzSRUkIobi/rustc_testrunner_tmpdir_reporting.K2C7radLLRcJ/mvce.rs:4:54: 4:60 (#5), covspan.span=/tmp/icemaker_global_tempdir.X6mzSRUkIobi/rustc_testrunner_tmpdir_reporting.K2C7radLLRcJ/mvce.rs:4:56: 4:58 (#6)
stack backtrace:
   0:     0x7f8862e267c3 - std::backtrace_rs::backtrace::libunwind::trace::h2ae476b6f0486ced
                               at /rustc/a0f398e89df9767c93c81cd58d44fdba071258a8/library/std/src/../../backtrace/src/backtrace/libunwind.rs:117:9
   1:     0x7f8862e267c3 - std::backtrace_rs::backtrace::trace_unsynchronized::h8c0216456ec4f735
                               at /rustc/a0f398e89df9767c93c81cd58d44fdba071258a8/library/std/src/../../backtrace/src/backtrace/mod.rs:66:14
   2:     0x7f8862e267c3 - std::sys::backtrace::_print_fmt::h5ff9e60cfd308fa7
                               at /rustc/a0f398e89df9767c93c81cd58d44fdba071258a8/library/std/src/sys/backtrace.rs:66:9
   3:     0x7f8862e267c3 - <std::sys::backtrace::BacktraceLock::print::DisplayBacktrace as core::fmt::Display>::fmt::hab094fc575a7b1a6
                               at /rustc/a0f398e89df9767c93c81cd58d44fdba071258a8/library/std/src/sys/backtrace.rs:39:26
   4:     0x7f885e1eb982 - core::fmt::rt::Argument::fmt::hd2d1999c4c43ab4b
                               at /rustc/a0f398e89df9767c93c81cd58d44fdba071258a8/library/core/src/fmt/rt.rs:173:76
   5:     0x7f885e1eb982 - core::fmt::write::h223cf3f5884c69e3
                               at /rustc/a0f398e89df9767c93c81cd58d44fdba071258a8/library/core/src/fmt/mod.rs:1468:25
   6:     0x7f8862db29c9 - std::io::default_write_fmt::hd6579fe41db0cf38
                               at /rustc/a0f398e89df9767c93c81cd58d44fdba071258a8/library/std/src/io/mod.rs:639:11
   7:     0x7f8862db29c9 - std::io::Write::write_fmt::h7a1b62a3f2a5ce82
                               at /rustc/a0f398e89df9767c93c81cd58d44fdba071258a8/library/std/src/io/mod.rs:1954:13
   8:     0x7f8862dd789b - std::sys::backtrace::BacktraceLock::print::h801a8b54e8af94e8
                               at /rustc/a0f398e89df9767c93c81cd58d44fdba071258a8/library/std/src/sys/backtrace.rs:42:9
   9:     0x7f8862dd789b - std::panicking::default_hook::{{closure}}::h2cf9f3b91d21ef19
  10:     0x7f8862dd7435 - std::panicking::default_hook::h9842b61a4eeedde9
                               at /rustc/a0f398e89df9767c93c81cd58d44fdba071258a8/library/std/src/panicking.rs:328:9
  11:     0x7f885f5eebc2 - <alloc[af02704fa02b2818]::boxed::Box<dyn for<'a, 'b> core[d026e51bcd835cda]::ops::function::Fn<(&'a std[ca93dc60a66752d0]::panic::PanicHookInfo<'b>,), Output = ()> + core[d026e51bcd835cda]::marker::Send + core[d026e51bcd835cda]::marker::Sync> as core[d026e51bcd835cda]::ops::function::Fn<(&std[ca93dc60a66752d0]::panic::PanicHookInfo,)>>::call
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/library/alloc/src/boxed.rs:2018:9
  12:     0x7f885f5eebc2 - rustc_driver_impl[c3970283c55b1ffe]::install_ice_hook::{closure#1}
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_driver_impl/src/lib.rs:1431:17
  13:     0x7f885f5eebc2 - <alloc[af02704fa02b2818]::boxed::Box<rustc_driver_impl[c3970283c55b1ffe]::install_ice_hook::{closure#1}> as core[d026e51bcd835cda]::ops::function::Fn<(&dyn for<'a, 'b> core[d026e51bcd835cda]::ops::function::Fn<(&'a std[ca93dc60a66752d0]::panic::PanicHookInfo<'b>,), Output = ()> + core[d026e51bcd835cda]::marker::Send + core[d026e51bcd835cda]::marker::Sync, &std[ca93dc60a66752d0]::panic::PanicHookInfo)>>::call
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/library/alloc/src/boxed.rs:2018:9
  14:     0x7f885f5eebc2 - std[ca93dc60a66752d0]::panicking::update_hook::<alloc[af02704fa02b2818]::boxed::Box<rustc_driver_impl[c3970283c55b1ffe]::install_ice_hook::{closure#1}>>::{closure#0}
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/library/std/src/panicking.rs:243:47
  15:     0x7f8862dd7eb4 - <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call::h3629b8cb2bc0b7fd
                               at /rustc/a0f398e89df9767c93c81cd58d44fdba071258a8/library/alloc/src/boxed.rs:2018:9
  16:     0x7f8862dd7eb4 - std::panicking::panic_with_hook::hc1e06896da5536d6
                               at /rustc/a0f398e89df9767c93c81cd58d44fdba071258a8/library/std/src/panicking.rs:842:13
  17:     0x7f8862dd7b5c - std::panicking::panic_handler::{{closure}}::h1bf52cb896e623c4
                               at /rustc/a0f398e89df9767c93c81cd58d44fdba071258a8/library/std/src/panicking.rs:707:13
  18:     0x7f8862dce699 - std::sys::backtrace::__rust_end_short_backtrace::hf0a45c7a9e22e84f
                               at /rustc/a0f398e89df9767c93c81cd58d44fdba071258a8/library/std/src/sys/backtrace.rs:174:18
  19:     0x7f8862da17bd - __rustc[5006c3107915356d]::rust_begin_unwind
                               at /rustc/a0f398e89df9767c93c81cd58d44fdba071258a8/library/std/src/panicking.rs:698:5
  20:     0x7f885e1fc140 - core::panicking::panic_fmt::h092cd596083e1fd0
                               at /rustc/a0f398e89df9767c93c81cd58d44fdba071258a8/library/core/src/panicking.rs:78:14
  21:     0x7f8861173182 - rustc_mir_transform[dd88981c0cc54aa9]::coverage::spans::extract_refined_covspans::{closure#1}
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_mir_transform/src/coverage/spans.rs:56:13
  22:     0x7f8861258971 - rustc_mir_transform[dd88981c0cc54aa9]::coverage::spans::extract_refined_covspans
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_mir_transform/src/coverage/spans.rs:68:13
  23:     0x7f886122f2e0 - rustc_mir_transform[dd88981c0cc54aa9]::coverage::mappings::extract_mappings_from_mir
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_mir_transform/src/coverage/mappings.rs:29:5
  24:     0x7f886122f2e0 - rustc_mir_transform[dd88981c0cc54aa9]::coverage::instrument_function_for_coverage
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_mir_transform/src/coverage/mod.rs:76:9
  25:     0x7f886120b4a7 - rustc_mir_transform[dd88981c0cc54aa9]::pass_manager::run_passes_inner
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_mir_transform/src/pass_manager.rs:308:22
  26:     0x7f8861181aca - rustc_mir_transform[dd88981c0cc54aa9]::pass_manager::run_passes
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_mir_transform/src/pass_manager.rs:210:5
  27:     0x7f8861181aca - rustc_mir_transform[dd88981c0cc54aa9]::mir_promoted
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_mir_transform/src/lib.rs:446:5
  28:     0x7f8861aa449c - rustc_query_impl[e8126b9c91291e43]::query_impl::mir_promoted::dynamic_query::{closure#2}::{closure#0}
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_query_impl/src/plumbing.rs:294:9
  29:     0x7f8861aa449c - rustc_query_impl[e8126b9c91291e43]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[e8126b9c91291e43]::query_impl::mir_promoted::dynamic_query::{closure#2}::{closure#0}, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>>
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_query_impl/src/plumbing.rs:578:18
  30:     0x7f8861ebfe97 - rustc_query_impl[e8126b9c91291e43]::query_impl::mir_promoted::dynamic_query::{closure#2}
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_query_impl/src/plumbing.rs:655:25
  31:     0x7f8861ebfe97 - <rustc_query_impl[e8126b9c91291e43]::query_impl::mir_promoted::dynamic_query::{closure#2} as core[d026e51bcd835cda]::ops::function::FnOnce<(rustc_middle[686fa7803823022e]::ty::context::TyCtxt, rustc_span[4f3ba1db772fcaf2]::def_id::LocalDefId)>>::call_once
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/library/core/src/ops/function.rs:250:5
  32:     0x7f8861babd61 - <rustc_query_impl[e8126b9c91291e43]::DynamicConfig<rustc_data_structures[3f38cd30b21b0412]::vec_cache::VecCache<rustc_span[4f3ba1db772fcaf2]::def_id::LocalDefId, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[aae3d10a6d329076]::dep_graph::graph::DepNodeIndex>, false, false, false> as rustc_query_system[aae3d10a6d329076]::query::config::QueryConfig<rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt>>::compute
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_query_impl/src/lib.rs:122:9
  33:     0x7f8861babd61 - rustc_query_system[aae3d10a6d329076]::query::plumbing::execute_job_non_incr::<rustc_query_impl[e8126b9c91291e43]::DynamicConfig<rustc_data_structures[3f38cd30b21b0412]::vec_cache::VecCache<rustc_span[4f3ba1db772fcaf2]::def_id::LocalDefId, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[aae3d10a6d329076]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt>::{closure#0}
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_query_system/src/query/plumbing.rs:493:72
  34:     0x7f8861babd61 - rustc_middle[686fa7803823022e]::ty::context::tls::enter_context::<rustc_query_system[aae3d10a6d329076]::query::plumbing::execute_job_non_incr<rustc_query_impl[e8126b9c91291e43]::DynamicConfig<rustc_data_structures[3f38cd30b21b0412]::vec_cache::VecCache<rustc_span[4f3ba1db772fcaf2]::def_id::LocalDefId, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[aae3d10a6d329076]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>>::{closure#0}
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_middle/src/ty/context/tls.rs:60:9
  35:     0x7f8861babd61 - <std[ca93dc60a66752d0]::thread::local::LocalKey<core[d026e51bcd835cda]::cell::Cell<*const ()>>>::try_with::<rustc_middle[686fa7803823022e]::ty::context::tls::enter_context<rustc_query_system[aae3d10a6d329076]::query::plumbing::execute_job_non_incr<rustc_query_impl[e8126b9c91291e43]::DynamicConfig<rustc_data_structures[3f38cd30b21b0412]::vec_cache::VecCache<rustc_span[4f3ba1db772fcaf2]::def_id::LocalDefId, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[aae3d10a6d329076]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>>::{closure#0}, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>>
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/library/std/src/thread/local.rs:508:12
  36:     0x7f8861babd61 - <std[ca93dc60a66752d0]::thread::local::LocalKey<core[d026e51bcd835cda]::cell::Cell<*const ()>>>::with::<rustc_middle[686fa7803823022e]::ty::context::tls::enter_context<rustc_query_system[aae3d10a6d329076]::query::plumbing::execute_job_non_incr<rustc_query_impl[e8126b9c91291e43]::DynamicConfig<rustc_data_structures[3f38cd30b21b0412]::vec_cache::VecCache<rustc_span[4f3ba1db772fcaf2]::def_id::LocalDefId, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[aae3d10a6d329076]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>>::{closure#0}, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>>
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/library/std/src/thread/local.rs:472:20
  37:     0x7f8861babd61 - rustc_middle[686fa7803823022e]::ty::context::tls::enter_context::<rustc_query_system[aae3d10a6d329076]::query::plumbing::execute_job_non_incr<rustc_query_impl[e8126b9c91291e43]::DynamicConfig<rustc_data_structures[3f38cd30b21b0412]::vec_cache::VecCache<rustc_span[4f3ba1db772fcaf2]::def_id::LocalDefId, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[aae3d10a6d329076]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>>
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_middle/src/ty/context/tls.rs:57:9
  38:     0x7f8861babd61 - <rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt as rustc_query_system[aae3d10a6d329076]::query::QueryContext>::start_query::<rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[aae3d10a6d329076]::query::plumbing::execute_job_non_incr<rustc_query_impl[e8126b9c91291e43]::DynamicConfig<rustc_data_structures[3f38cd30b21b0412]::vec_cache::VecCache<rustc_span[4f3ba1db772fcaf2]::def_id::LocalDefId, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[aae3d10a6d329076]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_query_impl/src/plumbing.rs:161:13
  39:     0x7f8861babd61 - rustc_middle[686fa7803823022e]::ty::context::tls::with_related_context::<<rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt as rustc_query_system[aae3d10a6d329076]::query::QueryContext>::start_query<rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[aae3d10a6d329076]::query::plumbing::execute_job_non_incr<rustc_query_impl[e8126b9c91291e43]::DynamicConfig<rustc_data_structures[3f38cd30b21b0412]::vec_cache::VecCache<rustc_span[4f3ba1db772fcaf2]::def_id::LocalDefId, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[aae3d10a6d329076]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>>::{closure#0}
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_middle/src/ty/context/tls.rs:112:9
  40:     0x7f8861babd61 - rustc_middle[686fa7803823022e]::ty::context::tls::with_context::<rustc_middle[686fa7803823022e]::ty::context::tls::with_related_context<<rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt as rustc_query_system[aae3d10a6d329076]::query::QueryContext>::start_query<rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[aae3d10a6d329076]::query::plumbing::execute_job_non_incr<rustc_query_impl[e8126b9c91291e43]::DynamicConfig<rustc_data_structures[3f38cd30b21b0412]::vec_cache::VecCache<rustc_span[4f3ba1db772fcaf2]::def_id::LocalDefId, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[aae3d10a6d329076]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>>::{closure#0}, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>>::{closure#0}
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_middle/src/ty/context/tls.rs:90:36
  41:     0x7f8861babd61 - rustc_middle[686fa7803823022e]::ty::context::tls::with_context_opt::<rustc_middle[686fa7803823022e]::ty::context::tls::with_context<rustc_middle[686fa7803823022e]::ty::context::tls::with_related_context<<rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt as rustc_query_system[aae3d10a6d329076]::query::QueryContext>::start_query<rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[aae3d10a6d329076]::query::plumbing::execute_job_non_incr<rustc_query_impl[e8126b9c91291e43]::DynamicConfig<rustc_data_structures[3f38cd30b21b0412]::vec_cache::VecCache<rustc_span[4f3ba1db772fcaf2]::def_id::LocalDefId, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[aae3d10a6d329076]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>>::{closure#0}, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>>::{closure#0}, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>>
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_middle/src/ty/context/tls.rs:79:18
  42:     0x7f8861babd61 - rustc_middle[686fa7803823022e]::ty::context::tls::with_context::<rustc_middle[686fa7803823022e]::ty::context::tls::with_related_context<<rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt as rustc_query_system[aae3d10a6d329076]::query::QueryContext>::start_query<rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[aae3d10a6d329076]::query::plumbing::execute_job_non_incr<rustc_query_impl[e8126b9c91291e43]::DynamicConfig<rustc_data_structures[3f38cd30b21b0412]::vec_cache::VecCache<rustc_span[4f3ba1db772fcaf2]::def_id::LocalDefId, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[aae3d10a6d329076]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>>::{closure#0}, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>>
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_middle/src/ty/context/tls.rs:90:5
  43:     0x7f8861babd61 - rustc_middle[686fa7803823022e]::ty::context::tls::with_related_context::<<rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt as rustc_query_system[aae3d10a6d329076]::query::QueryContext>::start_query<rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[aae3d10a6d329076]::query::plumbing::execute_job_non_incr<rustc_query_impl[e8126b9c91291e43]::DynamicConfig<rustc_data_structures[3f38cd30b21b0412]::vec_cache::VecCache<rustc_span[4f3ba1db772fcaf2]::def_id::LocalDefId, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[aae3d10a6d329076]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>>
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_middle/src/ty/context/tls.rs:103:5
  44:     0x7f8861babd61 - <rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt as rustc_query_system[aae3d10a6d329076]::query::QueryContext>::start_query::<rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[aae3d10a6d329076]::query::plumbing::execute_job_non_incr<rustc_query_impl[e8126b9c91291e43]::DynamicConfig<rustc_data_structures[3f38cd30b21b0412]::vec_cache::VecCache<rustc_span[4f3ba1db772fcaf2]::def_id::LocalDefId, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[aae3d10a6d329076]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt>::{closure#0}>
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_query_impl/src/plumbing.rs:147:9
  45:     0x7f8861babd61 - rustc_query_system[aae3d10a6d329076]::query::plumbing::execute_job_non_incr::<rustc_query_impl[e8126b9c91291e43]::DynamicConfig<rustc_data_structures[3f38cd30b21b0412]::vec_cache::VecCache<rustc_span[4f3ba1db772fcaf2]::def_id::LocalDefId, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[aae3d10a6d329076]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt>
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_query_system/src/query/plumbing.rs:493:22
  46:     0x7f8861babd61 - rustc_query_system[aae3d10a6d329076]::query::plumbing::execute_job::<rustc_query_impl[e8126b9c91291e43]::DynamicConfig<rustc_data_structures[3f38cd30b21b0412]::vec_cache::VecCache<rustc_span[4f3ba1db772fcaf2]::def_id::LocalDefId, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[aae3d10a6d329076]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt, false>
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_query_system/src/query/plumbing.rs:429:9
  47:     0x7f8861babd61 - rustc_query_system[aae3d10a6d329076]::query::plumbing::try_execute_query::<rustc_query_impl[e8126b9c91291e43]::DynamicConfig<rustc_data_structures[3f38cd30b21b0412]::vec_cache::VecCache<rustc_span[4f3ba1db772fcaf2]::def_id::LocalDefId, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[aae3d10a6d329076]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt, false>
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_query_system/src/query/plumbing.rs:372:13
  48:     0x7f8861ce4a8f - rustc_query_system[aae3d10a6d329076]::query::plumbing::get_query_non_incr::<rustc_query_impl[e8126b9c91291e43]::DynamicConfig<rustc_data_structures[3f38cd30b21b0412]::vec_cache::VecCache<rustc_span[4f3ba1db772fcaf2]::def_id::LocalDefId, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[aae3d10a6d329076]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt>::{closure#0}
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_query_system/src/query/plumbing.rs:814:32
  49:     0x7f8861ce4a8f - stacker[5eef17c692f1bb0b]::maybe_grow::<rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[aae3d10a6d329076]::query::plumbing::get_query_non_incr<rustc_query_impl[e8126b9c91291e43]::DynamicConfig<rustc_data_structures[3f38cd30b21b0412]::vec_cache::VecCache<rustc_span[4f3ba1db772fcaf2]::def_id::LocalDefId, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[aae3d10a6d329076]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt>::{closure#0}>
                               at /rust/deps/stacker-0.1.21/src/lib.rs:57:9
  50:     0x7f8861ce4a8f - rustc_data_structures[3f38cd30b21b0412]::stack::ensure_sufficient_stack::<rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[aae3d10a6d329076]::query::plumbing::get_query_non_incr<rustc_query_impl[e8126b9c91291e43]::DynamicConfig<rustc_data_structures[3f38cd30b21b0412]::vec_cache::VecCache<rustc_span[4f3ba1db772fcaf2]::def_id::LocalDefId, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[aae3d10a6d329076]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt>::{closure#0}>
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_data_structures/src/stack.rs:21:5
  51:     0x7f8861ce4a8f - rustc_query_system[aae3d10a6d329076]::query::plumbing::get_query_non_incr::<rustc_query_impl[e8126b9c91291e43]::DynamicConfig<rustc_data_structures[3f38cd30b21b0412]::vec_cache::VecCache<rustc_span[4f3ba1db772fcaf2]::def_id::LocalDefId, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[aae3d10a6d329076]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt>
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_query_system/src/query/plumbing.rs:814:5
  52:     0x7f8861ce4a8f - rustc_query_impl[e8126b9c91291e43]::query_impl::mir_promoted::get_query_non_incr::__rust_end_short_backtrace
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_query_impl/src/plumbing.rs:631:26
  53:     0x7f885eb23ceb - rustc_middle[686fa7803823022e]::query::plumbing::query_get_at::<rustc_data_structures[3f38cd30b21b0412]::vec_cache::VecCache<rustc_span[4f3ba1db772fcaf2]::def_id::LocalDefId, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[aae3d10a6d329076]::dep_graph::graph::DepNodeIndex>>
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_middle/src/query/plumbing.rs:182:17
  54:     0x7f885eb23ceb - <rustc_middle[686fa7803823022e]::query::plumbing::TyCtxtAt>::mir_promoted::<rustc_span[4f3ba1db772fcaf2]::def_id::LocalDefId>
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_middle/src/query/plumbing.rs:475:31
  55:     0x7f885eb23ceb - <rustc_middle[686fa7803823022e]::ty::context::TyCtxt>::mir_promoted::<rustc_span[4f3ba1db772fcaf2]::def_id::LocalDefId>
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_middle/src/query/plumbing.rs:466:35
  56:     0x7f885eb23ceb - rustc_borrowck[61f5c28638a576c]::mir_borrowck
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_borrowck/src/lib.rs:124:31
  57:     0x7f8861aa411e - rustc_query_impl[e8126b9c91291e43]::query_impl::mir_borrowck::dynamic_query::{closure#2}::{closure#0}
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_query_impl/src/plumbing.rs:294:9
  58:     0x7f8861aa411e - rustc_query_impl[e8126b9c91291e43]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[e8126b9c91291e43]::query_impl::mir_borrowck::dynamic_query::{closure#2}::{closure#0}, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 8usize]>>
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_query_impl/src/plumbing.rs:578:18
  59:     0x7f8861ebf17f - rustc_query_impl[e8126b9c91291e43]::query_impl::mir_borrowck::dynamic_query::{closure#2}
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_query_impl/src/plumbing.rs:655:25
  60:     0x7f8861ebf17f - <rustc_query_impl[e8126b9c91291e43]::query_impl::mir_borrowck::dynamic_query::{closure#2} as core[d026e51bcd835cda]::ops::function::FnOnce<(rustc_middle[686fa7803823022e]::ty::context::TyCtxt, rustc_span[4f3ba1db772fcaf2]::def_id::LocalDefId)>>::call_once
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/library/core/src/ops/function.rs:250:5
  61:     0x7f8861ba693e - <rustc_query_impl[e8126b9c91291e43]::DynamicConfig<rustc_data_structures[3f38cd30b21b0412]::vec_cache::VecCache<rustc_hir_id[b9d561e38648682f]::OwnerId, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[aae3d10a6d329076]::dep_graph::graph::DepNodeIndex>, false, false, false> as rustc_query_system[aae3d10a6d329076]::query::config::QueryConfig<rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt>>::compute
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_query_impl/src/lib.rs:122:9
  62:     0x7f8861ba693e - rustc_query_system[aae3d10a6d329076]::query::plumbing::execute_job_non_incr::<rustc_query_impl[e8126b9c91291e43]::DynamicConfig<rustc_data_structures[3f38cd30b21b0412]::vec_cache::VecCache<rustc_hir_id[b9d561e38648682f]::OwnerId, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[aae3d10a6d329076]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt>::{closure#0}
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_query_system/src/query/plumbing.rs:493:72
  63:     0x7f8861ba693e - rustc_middle[686fa7803823022e]::ty::context::tls::enter_context::<rustc_query_system[aae3d10a6d329076]::query::plumbing::execute_job_non_incr<rustc_query_impl[e8126b9c91291e43]::DynamicConfig<rustc_data_structures[3f38cd30b21b0412]::vec_cache::VecCache<rustc_hir_id[b9d561e38648682f]::OwnerId, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[aae3d10a6d329076]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 8usize]>>::{closure#0}
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_middle/src/ty/context/tls.rs:60:9
  64:     0x7f8861ba693e - <std[ca93dc60a66752d0]::thread::local::LocalKey<core[d026e51bcd835cda]::cell::Cell<*const ()>>>::try_with::<rustc_middle[686fa7803823022e]::ty::context::tls::enter_context<rustc_query_system[aae3d10a6d329076]::query::plumbing::execute_job_non_incr<rustc_query_impl[e8126b9c91291e43]::DynamicConfig<rustc_data_structures[3f38cd30b21b0412]::vec_cache::VecCache<rustc_hir_id[b9d561e38648682f]::OwnerId, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[aae3d10a6d329076]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 8usize]>>
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/library/std/src/thread/local.rs:508:12
  65:     0x7f8861ba693e - <std[ca93dc60a66752d0]::thread::local::LocalKey<core[d026e51bcd835cda]::cell::Cell<*const ()>>>::with::<rustc_middle[686fa7803823022e]::ty::context::tls::enter_context<rustc_query_system[aae3d10a6d329076]::query::plumbing::execute_job_non_incr<rustc_query_impl[e8126b9c91291e43]::DynamicConfig<rustc_data_structures[3f38cd30b21b0412]::vec_cache::VecCache<rustc_hir_id[b9d561e38648682f]::OwnerId, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[aae3d10a6d329076]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 8usize]>>
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/library/std/src/thread/local.rs:472:20
  66:     0x7f8861ba693e - rustc_middle[686fa7803823022e]::ty::context::tls::enter_context::<rustc_query_system[aae3d10a6d329076]::query::plumbing::execute_job_non_incr<rustc_query_impl[e8126b9c91291e43]::DynamicConfig<rustc_data_structures[3f38cd30b21b0412]::vec_cache::VecCache<rustc_hir_id[b9d561e38648682f]::OwnerId, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[aae3d10a6d329076]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 8usize]>>
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_middle/src/ty/context/tls.rs:57:9
  67:     0x7f8861ba693e - <rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt as rustc_query_system[aae3d10a6d329076]::query::QueryContext>::start_query::<rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[aae3d10a6d329076]::query::plumbing::execute_job_non_incr<rustc_query_impl[e8126b9c91291e43]::DynamicConfig<rustc_data_structures[3f38cd30b21b0412]::vec_cache::VecCache<rustc_hir_id[b9d561e38648682f]::OwnerId, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[aae3d10a6d329076]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_query_impl/src/plumbing.rs:161:13
  68:     0x7f8861ba693e - rustc_middle[686fa7803823022e]::ty::context::tls::with_related_context::<<rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt as rustc_query_system[aae3d10a6d329076]::query::QueryContext>::start_query<rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[aae3d10a6d329076]::query::plumbing::execute_job_non_incr<rustc_query_impl[e8126b9c91291e43]::DynamicConfig<rustc_data_structures[3f38cd30b21b0412]::vec_cache::VecCache<rustc_hir_id[b9d561e38648682f]::OwnerId, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[aae3d10a6d329076]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 8usize]>>::{closure#0}
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_middle/src/ty/context/tls.rs:112:9
  69:     0x7f8861ba693e - rustc_middle[686fa7803823022e]::ty::context::tls::with_context::<rustc_middle[686fa7803823022e]::ty::context::tls::with_related_context<<rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt as rustc_query_system[aae3d10a6d329076]::query::QueryContext>::start_query<rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[aae3d10a6d329076]::query::plumbing::execute_job_non_incr<rustc_query_impl[e8126b9c91291e43]::DynamicConfig<rustc_data_structures[3f38cd30b21b0412]::vec_cache::VecCache<rustc_hir_id[b9d561e38648682f]::OwnerId, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[aae3d10a6d329076]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 8usize]>>::{closure#0}
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_middle/src/ty/context/tls.rs:90:36
  70:     0x7f8861ba693e - rustc_middle[686fa7803823022e]::ty::context::tls::with_context_opt::<rustc_middle[686fa7803823022e]::ty::context::tls::with_context<rustc_middle[686fa7803823022e]::ty::context::tls::with_related_context<<rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt as rustc_query_system[aae3d10a6d329076]::query::QueryContext>::start_query<rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[aae3d10a6d329076]::query::plumbing::execute_job_non_incr<rustc_query_impl[e8126b9c91291e43]::DynamicConfig<rustc_data_structures[3f38cd30b21b0412]::vec_cache::VecCache<rustc_hir_id[b9d561e38648682f]::OwnerId, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[aae3d10a6d329076]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 8usize]>>
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_middle/src/ty/context/tls.rs:79:18
  71:     0x7f8861ba693e - rustc_middle[686fa7803823022e]::ty::context::tls::with_context::<rustc_middle[686fa7803823022e]::ty::context::tls::with_related_context<<rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt as rustc_query_system[aae3d10a6d329076]::query::QueryContext>::start_query<rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[aae3d10a6d329076]::query::plumbing::execute_job_non_incr<rustc_query_impl[e8126b9c91291e43]::DynamicConfig<rustc_data_structures[3f38cd30b21b0412]::vec_cache::VecCache<rustc_hir_id[b9d561e38648682f]::OwnerId, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[aae3d10a6d329076]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 8usize]>>
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_middle/src/ty/context/tls.rs:90:5
  72:     0x7f8861ba693e - rustc_middle[686fa7803823022e]::ty::context::tls::with_related_context::<<rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt as rustc_query_system[aae3d10a6d329076]::query::QueryContext>::start_query<rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[aae3d10a6d329076]::query::plumbing::execute_job_non_incr<rustc_query_impl[e8126b9c91291e43]::DynamicConfig<rustc_data_structures[3f38cd30b21b0412]::vec_cache::VecCache<rustc_hir_id[b9d561e38648682f]::OwnerId, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[aae3d10a6d329076]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 8usize]>>
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_middle/src/ty/context/tls.rs:103:5
  73:     0x7f8861ba693e - <rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt as rustc_query_system[aae3d10a6d329076]::query::QueryContext>::start_query::<rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[aae3d10a6d329076]::query::plumbing::execute_job_non_incr<rustc_query_impl[e8126b9c91291e43]::DynamicConfig<rustc_data_structures[3f38cd30b21b0412]::vec_cache::VecCache<rustc_hir_id[b9d561e38648682f]::OwnerId, rustc_middle[686fa7803823022e]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[aae3d10a6d329076]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[e8126b9c91291e43]::plumbing::QueryCtxt>::{closure#0}>
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_query_impl/src/plumbing.rs:147:9

<snip>


 139:     0x7f885f5f924b - rustc_interface[bbcd5ca1d39a7461]::passes::create_and_enter_global_ctxt::<core[d026e51bcd835cda]::option::Option<rustc_interface[bbcd5ca1d39a7461]::queries::Linker>, rustc_driver_impl[c3970283c55b1ffe]::run_compiler::{closure#0}::{closure#2}>::{closure#2}
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_interface/src/passes.rs:983:9
 140:     0x7f885f5f924b - <rustc_interface[bbcd5ca1d39a7461]::passes::create_and_enter_global_ctxt<core[d026e51bcd835cda]::option::Option<rustc_interface[bbcd5ca1d39a7461]::queries::Linker>, rustc_driver_impl[c3970283c55b1ffe]::run_compiler::{closure#0}::{closure#2}>::{closure#2} as core[d026e51bcd835cda]::ops::function::FnOnce<(&rustc_session[850908e2ad126089]::session::Session, rustc_middle[686fa7803823022e]::ty::context::CurrentGcx, alloc[af02704fa02b2818]::sync::Arc<rustc_data_structures[3f38cd30b21b0412]::jobserver::Proxy>, &std[ca93dc60a66752d0]::sync::once_lock::OnceLock<rustc_middle[686fa7803823022e]::ty::context::GlobalCtxt>, &rustc_data_structures[3f38cd30b21b0412]::sync::worker_local::WorkerLocal<rustc_middle[686fa7803823022e]::arena::Arena>, &rustc_data_structures[3f38cd30b21b0412]::sync::worker_local::WorkerLocal<rustc_hir[90b51a211c376f75]::Arena>, rustc_driver_impl[c3970283c55b1ffe]::run_compiler::{closure#0}::{closure#2})>>::call_once::{shim:vtable#0}
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/library/core/src/ops/function.rs:250:5
 141:     0x7f885f56fd40 - <alloc[af02704fa02b2818]::boxed::Box<dyn for<'a> core[d026e51bcd835cda]::ops::function::FnOnce<(&'a rustc_session[850908e2ad126089]::session::Session, rustc_middle[686fa7803823022e]::ty::context::CurrentGcx, alloc[af02704fa02b2818]::sync::Arc<rustc_data_structures[3f38cd30b21b0412]::jobserver::Proxy>, &'a std[ca93dc60a66752d0]::sync::once_lock::OnceLock<rustc_middle[686fa7803823022e]::ty::context::GlobalCtxt<'a>>, &'a rustc_data_structures[3f38cd30b21b0412]::sync::worker_local::WorkerLocal<rustc_middle[686fa7803823022e]::arena::Arena<'a>>, &'a rustc_data_structures[3f38cd30b21b0412]::sync::worker_local::WorkerLocal<rustc_hir[90b51a211c376f75]::Arena<'a>>, rustc_driver_impl[c3970283c55b1ffe]::run_compiler::{closure#0}::{closure#2}), Output = core[d026e51bcd835cda]::option::Option<rustc_interface[bbcd5ca1d39a7461]::queries::Linker>>> as core[d026e51bcd835cda]::ops::function::FnOnce<(&rustc_session[850908e2ad126089]::session::Session, rustc_middle[686fa7803823022e]::ty::context::CurrentGcx, alloc[af02704fa02b2818]::sync::Arc<rustc_data_structures[3f38cd30b21b0412]::jobserver::Proxy>, &std[ca93dc60a66752d0]::sync::once_lock::OnceLock<rustc_middle[686fa7803823022e]::ty::context::GlobalCtxt>, &rustc_data_structures[3f38cd30b21b0412]::sync::worker_local::WorkerLocal<rustc_middle[686fa7803823022e]::arena::Arena>, &rustc_data_structures[3f38cd30b21b0412]::sync::worker_local::WorkerLocal<rustc_hir[90b51a211c376f75]::Arena>, rustc_driver_impl[c3970283c55b1ffe]::run_compiler::{closure#0}::{closure#2})>>::call_once
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/library/alloc/src/boxed.rs:2004:9
 142:     0x7f885f56fd40 - rustc_interface[bbcd5ca1d39a7461]::passes::create_and_enter_global_ctxt::<core[d026e51bcd835cda]::option::Option<rustc_interface[bbcd5ca1d39a7461]::queries::Linker>, rustc_driver_impl[c3970283c55b1ffe]::run_compiler::{closure#0}::{closure#2}>
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_interface/src/passes.rs:1024:5
 143:     0x7f885f5ec917 - rustc_driver_impl[c3970283c55b1ffe]::run_compiler::{closure#0}
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_driver_impl/src/lib.rs:344:22
 144:     0x7f885f5ec917 - rustc_interface[bbcd5ca1d39a7461]::interface::run_compiler::<(), rustc_driver_impl[c3970283c55b1ffe]::run_compiler::{closure#0}>::{closure#1}::{closure#0}
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_interface/src/interface.rs:531:80
 145:     0x7f885f5ec917 - <core[d026e51bcd835cda]::panic::unwind_safe::AssertUnwindSafe<rustc_interface[bbcd5ca1d39a7461]::interface::run_compiler<(), rustc_driver_impl[c3970283c55b1ffe]::run_compiler::{closure#0}>::{closure#1}::{closure#0}> as core[d026e51bcd835cda]::ops::function::FnOnce<()>>::call_once
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/library/core/src/panic/unwind_safe.rs:274:9
 146:     0x7f885f5ec917 - std[ca93dc60a66752d0]::panicking::catch_unwind::do_call::<core[d026e51bcd835cda]::panic::unwind_safe::AssertUnwindSafe<rustc_interface[bbcd5ca1d39a7461]::interface::run_compiler<(), rustc_driver_impl[c3970283c55b1ffe]::run_compiler::{closure#0}>::{closure#1}::{closure#0}>, ()>
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/library/std/src/panicking.rs:590:40
 147:     0x7f885f5ec917 - std[ca93dc60a66752d0]::panicking::catch_unwind::<(), core[d026e51bcd835cda]::panic::unwind_safe::AssertUnwindSafe<rustc_interface[bbcd5ca1d39a7461]::interface::run_compiler<(), rustc_driver_impl[c3970283c55b1ffe]::run_compiler::{closure#0}>::{closure#1}::{closure#0}>>
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/library/std/src/panicking.rs:553:19
 148:     0x7f885f5ec917 - std[ca93dc60a66752d0]::panic::catch_unwind::<core[d026e51bcd835cda]::panic::unwind_safe::AssertUnwindSafe<rustc_interface[bbcd5ca1d39a7461]::interface::run_compiler<(), rustc_driver_impl[c3970283c55b1ffe]::run_compiler::{closure#0}>::{closure#1}::{closure#0}>, ()>
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/library/std/src/panic.rs:359:14
 149:     0x7f885f5ec917 - rustc_interface[bbcd5ca1d39a7461]::interface::run_compiler::<(), rustc_driver_impl[c3970283c55b1ffe]::run_compiler::{closure#0}>::{closure#1}
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_interface/src/interface.rs:531:23
 150:     0x7f885f5d8ea6 - rustc_interface[bbcd5ca1d39a7461]::util::run_in_thread_pool_with_globals::<rustc_interface[bbcd5ca1d39a7461]::interface::run_compiler<(), rustc_driver_impl[c3970283c55b1ffe]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_interface/src/util.rs:208:17
 151:     0x7f885f5d8ea6 - rustc_interface[bbcd5ca1d39a7461]::util::run_in_thread_with_globals::<rustc_interface[bbcd5ca1d39a7461]::util::run_in_thread_pool_with_globals<rustc_interface[bbcd5ca1d39a7461]::interface::run_compiler<(), rustc_driver_impl[c3970283c55b1ffe]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}::{closure#0}
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_interface/src/util.rs:162:24
 152:     0x7f885f5d8ea6 - <scoped_tls[108f4f64339ddd91]::ScopedKey<rustc_span[4f3ba1db772fcaf2]::SessionGlobals>>::set::<rustc_interface[bbcd5ca1d39a7461]::util::run_in_thread_with_globals<rustc_interface[bbcd5ca1d39a7461]::util::run_in_thread_pool_with_globals<rustc_interface[bbcd5ca1d39a7461]::interface::run_compiler<(), rustc_driver_impl[c3970283c55b1ffe]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}::{closure#0}, ()>
                               at /rust/deps/scoped-tls-1.0.1/src/lib.rs:137:9
 153:     0x7f885f5d8ea6 - rustc_span[4f3ba1db772fcaf2]::create_session_globals_then::<(), rustc_interface[bbcd5ca1d39a7461]::util::run_in_thread_with_globals<rustc_interface[bbcd5ca1d39a7461]::util::run_in_thread_pool_with_globals<rustc_interface[bbcd5ca1d39a7461]::interface::run_compiler<(), rustc_driver_impl[c3970283c55b1ffe]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}::{closure#0}>
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_span/src/lib.rs:146:21
 154:     0x7f885f5d8ea6 - rustc_interface[bbcd5ca1d39a7461]::util::run_in_thread_with_globals::<rustc_interface[bbcd5ca1d39a7461]::util::run_in_thread_pool_with_globals<rustc_interface[bbcd5ca1d39a7461]::interface::run_compiler<(), rustc_driver_impl[c3970283c55b1ffe]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/compiler/rustc_interface/src/util.rs:158:17
 155:     0x7f885f5d8ea6 - std[ca93dc60a66752d0]::sys::backtrace::__rust_begin_short_backtrace::<rustc_interface[bbcd5ca1d39a7461]::util::run_in_thread_with_globals<rustc_interface[bbcd5ca1d39a7461]::util::run_in_thread_pool_with_globals<rustc_interface[bbcd5ca1d39a7461]::interface::run_compiler<(), rustc_driver_impl[c3970283c55b1ffe]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/library/std/src/sys/backtrace.rs:158:18
 156:     0x7f885f5f6c00 - <std[ca93dc60a66752d0]::thread::Builder>::spawn_unchecked_::<rustc_interface[bbcd5ca1d39a7461]::util::run_in_thread_with_globals<rustc_interface[bbcd5ca1d39a7461]::util::run_in_thread_pool_with_globals<rustc_interface[bbcd5ca1d39a7461]::interface::run_compiler<(), rustc_driver_impl[c3970283c55b1ffe]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1}::{closure#0}
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/library/std/src/thread/mod.rs:562:17
 157:     0x7f885f5f6c00 - <core[d026e51bcd835cda]::panic::unwind_safe::AssertUnwindSafe<<std[ca93dc60a66752d0]::thread::Builder>::spawn_unchecked_<rustc_interface[bbcd5ca1d39a7461]::util::run_in_thread_with_globals<rustc_interface[bbcd5ca1d39a7461]::util::run_in_thread_pool_with_globals<rustc_interface[bbcd5ca1d39a7461]::interface::run_compiler<(), rustc_driver_impl[c3970283c55b1ffe]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1}::{closure#0}> as core[d026e51bcd835cda]::ops::function::FnOnce<()>>::call_once
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/library/core/src/panic/unwind_safe.rs:274:9
 158:     0x7f885f5f6c00 - std[ca93dc60a66752d0]::panicking::catch_unwind::do_call::<core[d026e51bcd835cda]::panic::unwind_safe::AssertUnwindSafe<<std[ca93dc60a66752d0]::thread::Builder>::spawn_unchecked_<rustc_interface[bbcd5ca1d39a7461]::util::run_in_thread_with_globals<rustc_interface[bbcd5ca1d39a7461]::util::run_in_thread_pool_with_globals<rustc_interface[bbcd5ca1d39a7461]::interface::run_compiler<(), rustc_driver_impl[c3970283c55b1ffe]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1}::{closure#0}>, ()>
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/library/std/src/panicking.rs:590:40
 159:     0x7f885f5f6c00 - std[ca93dc60a66752d0]::panicking::catch_unwind::<(), core[d026e51bcd835cda]::panic::unwind_safe::AssertUnwindSafe<<std[ca93dc60a66752d0]::thread::Builder>::spawn_unchecked_<rustc_interface[bbcd5ca1d39a7461]::util::run_in_thread_with_globals<rustc_interface[bbcd5ca1d39a7461]::util::run_in_thread_pool_with_globals<rustc_interface[bbcd5ca1d39a7461]::interface::run_compiler<(), rustc_driver_impl[c3970283c55b1ffe]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1}::{closure#0}>>
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/library/std/src/panicking.rs:553:19
 160:     0x7f885f5f6c00 - std[ca93dc60a66752d0]::panic::catch_unwind::<core[d026e51bcd835cda]::panic::unwind_safe::AssertUnwindSafe<<std[ca93dc60a66752d0]::thread::Builder>::spawn_unchecked_<rustc_interface[bbcd5ca1d39a7461]::util::run_in_thread_with_globals<rustc_interface[bbcd5ca1d39a7461]::util::run_in_thread_pool_with_globals<rustc_interface[bbcd5ca1d39a7461]::interface::run_compiler<(), rustc_driver_impl[c3970283c55b1ffe]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1}::{closure#0}>, ()>
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/library/std/src/panic.rs:359:14
 161:     0x7f885f5f6c00 - <std[ca93dc60a66752d0]::thread::Builder>::spawn_unchecked_::<rustc_interface[bbcd5ca1d39a7461]::util::run_in_thread_with_globals<rustc_interface[bbcd5ca1d39a7461]::util::run_in_thread_pool_with_globals<rustc_interface[bbcd5ca1d39a7461]::interface::run_compiler<(), rustc_driver_impl[c3970283c55b1ffe]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1}
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/library/std/src/thread/mod.rs:560:30
 162:     0x7f885f5f6c00 - <<std[ca93dc60a66752d0]::thread::Builder>::spawn_unchecked_<rustc_interface[bbcd5ca1d39a7461]::util::run_in_thread_with_globals<rustc_interface[bbcd5ca1d39a7461]::util::run_in_thread_pool_with_globals<rustc_interface[bbcd5ca1d39a7461]::interface::run_compiler<(), rustc_driver_impl[c3970283c55b1ffe]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1} as core[d026e51bcd835cda]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
                               at /rustc-dev/a0f398e89df9767c93c81cd58d44fdba071258a8/library/core/src/ops/function.rs:250:5
 163:     0x7f8862dc7b15 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::hb07cb51c89f96f42
                               at /rustc/a0f398e89df9767c93c81cd58d44fdba071258a8/library/alloc/src/boxed.rs:2004:9
 164:     0x7f8862dc7b15 - std::sys::thread::unix::Thread::new::thread_start::h52341e3b19d8f635
                               at /rustc/a0f398e89df9767c93c81cd58d44fdba071258a8/library/std/src/sys/thread/unix.rs:126:17
 165:     0x7f885c6969cb - <unknown>
 166:     0x7f885c71aa0c - <unknown>
 167:                0x0 - <unknown>

error: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: please make sure that you have updated to the latest nightly

note: rustc 1.92.0-nightly (a0f398e89 2025-10-04) running on x86_64-unknown-linux-gnu

note: compiler flags: -C instrument-coverage -Z dump-mir-dir=dir

query stack during panic:
#0 [mir_promoted] promoting constants in MIR for `a`
#1 [mir_borrowck] borrow-checking `a`
#2 [analysis] running analysis passes on this crate
end of query stack

<snip>

query stack during panic:
#0 [mir_promoted] promoting constants in MIR for `c`
#1 [mir_borrowck] borrow-checking `c`
#2 [analysis] running analysis passes on this crate
end of query stack
error: aborting due to 1 previous error; 3 warnings emitted

For more information about this error, try `rustc --explain E0601`.

```

</p>
</details>

--- a/compiler/rustc_mir_transform/src/coverage/spans.rs
+++ b/compiler/rustc_mir_transform/src/coverage/spans.rs
@@ -55,12 +55,10 @@ pub(super) fn extract_refined_covspans<'tcx>(
             }
 
             // Each pushed covspan should have the same context as the body span.
-            // If it somehow doesn't, discard the covspan, or panic in debug builds.
+            // If it somehow doesn't, discard the covspan.
             if !body_span.eq_ctxt(covspan_span) {
-                debug_assert!(
-                    false,
-                    "span context mismatch: body_span={body_span:?}, covspan.span={covspan_span:?}"
-                );
+                // FIXME(Zalathar): Investigate how and why this is triggered
+                // by `tests/coverage/macros/context-mismatch-issue-147339.rs`.
                 return false;
             }
 

--- a/tests/coverage/macros/context-mismatch-issue-147339.cov-map
+++ b/tests/coverage/macros/context-mismatch-issue-147339.cov-map
@@ -1,0 +1,40 @@
+Function name: context_mismatch_issue_147339::a (unused)
+Raw bytes (14): 0x[01, 01, 00, 02, 00, 0c, 27, 00, 35, 00, 00, 3b, 00, 3c]
+Number of files: 1
+- file 0 => $DIR/context-mismatch-issue-147339.rs
+Number of expressions: 0
+Number of file 0 mappings: 2
+- Code(Zero) at (prev + 12, 39) to (start + 0, 53)
+- Code(Zero) at (prev + 0, 59) to (start + 0, 60)
+Highest counter ID seen: (none)
+
+Function name: context_mismatch_issue_147339::b (unused)
+Raw bytes (14): 0x[01, 01, 00, 02, 00, 0c, 27, 00, 35, 00, 00, 3b, 00, 3c]
+Number of files: 1
+- file 0 => $DIR/context-mismatch-issue-147339.rs
+Number of expressions: 0
+Number of file 0 mappings: 2
+- Code(Zero) at (prev + 12, 39) to (start + 0, 53)
+- Code(Zero) at (prev + 0, 59) to (start + 0, 60)
+Highest counter ID seen: (none)
+
+Function name: context_mismatch_issue_147339::c (unused)
+Raw bytes (14): 0x[01, 01, 00, 02, 00, 0c, 27, 00, 35, 00, 00, 3b, 00, 3c]
+Number of files: 1
+- file 0 => $DIR/context-mismatch-issue-147339.rs
+Number of expressions: 0
+Number of file 0 mappings: 2
+- Code(Zero) at (prev + 12, 39) to (start + 0, 53)
+- Code(Zero) at (prev + 0, 59) to (start + 0, 60)
+Highest counter ID seen: (none)
+
+Function name: context_mismatch_issue_147339::main
+Raw bytes (14): 0x[01, 01, 00, 02, 01, 14, 01, 00, 0a, 01, 00, 0c, 00, 0d]
+Number of files: 1
+- file 0 => $DIR/context-mismatch-issue-147339.rs
+Number of expressions: 0
+Number of file 0 mappings: 2
+- Code(Counter(0)) at (prev + 20, 1) to (start + 0, 10)
+- Code(Counter(0)) at (prev + 0, 12) to (start + 0, 13)
+Highest counter ID seen: c0
+

--- a/tests/coverage/macros/context-mismatch-issue-147339.coverage
+++ b/tests/coverage/macros/context-mismatch-issue-147339.coverage
@@ -1,0 +1,28 @@
+   LL|       |//@ edition: 2024
+   LL|       |
+   LL|       |// These nested macro expansions were found to cause span refinement to produce
+   LL|       |// spans with a context that doesn't match the function body span, triggering
+   LL|       |// a defensive check that discards the span.
+   LL|       |//
+   LL|       |// Reported in <https://github.com/rust-lang/rust/issues/147339>.
+   LL|       |
+   LL|       |macro_rules! foo {
+   LL|       |    ($($m:ident $($f:ident $v:tt)+),*) => {
+   LL|       |        $($(macro_rules! $f { () => { $v } })+)*
+   LL|      0|        $(macro_rules! $m { () => { $(fn $f() -> i32 { $v })+ } })*
+  ------------------
+  | Unexecuted instantiation: context_mismatch_issue_147339::a
+  ------------------
+  | Unexecuted instantiation: context_mismatch_issue_147339::b
+  ------------------
+  | Unexecuted instantiation: context_mismatch_issue_147339::c
+  ------------------
+   LL|       |    }
+   LL|       |}
+   LL|       |
+   LL|       |foo!(m a 1 b 2, n c 3);
+   LL|       |m!();
+   LL|       |n!();
+   LL|       |
+   LL|      1|fn main() {}
+

--- a/tests/coverage/macros/context-mismatch-issue-147339.rs
+++ b/tests/coverage/macros/context-mismatch-issue-147339.rs
@@ -1,0 +1,20 @@
+//@ edition: 2024
+
+// These nested macro expansions were found to cause span refinement to produce
+// spans with a context that doesn't match the function body span, triggering
+// a defensive check that discards the span.
+//
+// Reported in <https://github.com/rust-lang/rust/issues/147339>.
+
+macro_rules! foo {
+    ($($m:ident $($f:ident $v:tt)+),*) => {
+        $($(macro_rules! $f { () => { $v } })+)*
+        $(macro_rules! $m { () => { $(fn $f() -> i32 { $v })+ } })*
+    }
+}
+
+foo!(m a 1 b 2, n c 3);
+m!();
+n!();
+
+fn main() {}

